ansible/roles/ocp4-workload-rhtr-iot-managed/tasks/provision_streams.yaml

```diff
@@ -0,0 +1,29 @@
+---
+- name: Evaluate {{ operators_project }} namespace if not exists 
+  k8s:
+    api_version: v1
+    kind: Namespace
+    name: '{{ operators_project }}'
+    state: present
+
+- name: Evaluate Operator Group
+  k8s:
+    api_version: operators.coreos.com/v1alpha2
+    kind: OperatorGroup
+    name: workshop-operators
+    namespace: '{{ operators_project }}'
+
+- name: Evaluate Streams Subscription
+  k8s:
+    state: present
+    resource_definition: "{{ lookup('template', 'streams-subscription.yaml.j2') }}"
+
+- name: Wait for Streams operator to install
```

Comment:
A CRD being present doesn't mean the operator is installed. you should look for the CSV being successfully deployed. Something like this:

```
- name: Wait until Operator is Ready
  k8s_info:
    api_version: operators.coreos.com/v1alpha1
    kind: ClusterServiceVersion
    name: my_csv
  register: r_my_csv
  retries: 30
  delay: 20
  until:
  - r_my_csv.resources | length > 0
  - r_my_csv.resources[0].status is defined
  - r_my_csv.resources[0].status.phase is defined
  - r_my_csv.resources[0].status.phase == "Succeeded"
```